File: app/src/main/java/com/amirreza/quizapplication/feature_authentication/presentation/fragment_login/LoginFragment.java
package com.amirreza.quizapplication.feature_authentication.presentation.fragment_login;

import android.app.FragmentManager;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.fragment.NavHostFragment;

import com.amirreza.quizeapplication.R;
import com.amirreza.quizeapplication.databinding.FragmentLoginBinding;


public class LoginFragment extends Fragment {
    private FragmentLoginBinding binding;




    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding =  FragmentLoginBinding.inflate(inflater,container,false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

//        NavHostFragment navHostFragment =
//                (NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment);
//
        NavController navController = Navigation.findNavController(view);

        binding.logInBtn.setOnClickListener(btn -> {

        });

        binding.signUpBtn.setOnClickListener(btn->{
            navController.navigate(R.id.signUpFragment);
        });
    }
}

File: app/src/main/java/com/amirreza/quizapplication/feature_authentication/presentation/fragment_login/LoginFragment.kt
package com.amirreza.quizapplication.feature_authentication.presentation.fragment_login

import androidx.navigation.Navigation.findNavController
import com.amirreza.quizapplication.util.base.QuizBaseFragment
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import androidx.navigation.fragment.findNavController
import com.amirreza.quizeapplication.R
import com.amirreza.quizeapplication.databinding.FragmentLoginBinding
import io.reactivex.CompletableObserver
import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import org.koin.android.ext.android.inject

class LoginFragment : QuizBaseFragment() {
    private lateinit var binding: FragmentLoginBinding
    private val viewModel:LoginViewModel by inject()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentLoginBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.logInBtn.setOnClickListener { btn: View ->
            val username = binding.usernameValue.text.toString()
            val password = binding.passwordValue.text.toString()

            viewModel.login(username,password)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object : CompletableObserver{
                    override fun onSubscribe(d: Disposable) {

                    }

                    override fun onComplete() {
                        showToast("Login was successful !!")
                        findNavController().navigate(R.id.action_loginFragment_to_homeFragment)
                    }

                    override fun onError(e: Throwable) {
                        showToast("Login encountered an error !!")
                    }

                })
        }
        binding.signUpBtn.setOnClickListener {
            findNavController().navigate(R.id.action_loginFragment_to_signUpFragment)
        }
    }
}